widgets/common/src/NodeFilter.tsx

```diff
@@ -133,15 +135,28 @@ export default class NodeFilter extends React.Component {
             .getManager()
             .doGet(fetchUrl, { params })
             .then(data => {
-                let ids = _.chain(data.items || [])
-                    .map(item => item.id)
-                    .uniqWith(_.isEqual)
-                    .value();
+                let allowedOptions = [];
```

Comment:
This file can freely use TypeScript that can make understanding the code easier :slightly_smiling_face: Thus, I suggest we sprinkle a bit of TS here:

```suggestion
                let allowedOptions: string[] = [];
```

also, we could add types in many other places - the function signature:

```ts
    fetchData(fetchUrl: string, params: any, optionsField: string) {
```

Reply:
Sure we could add it in whole of ts file, but the changes will be very big and unneeded regarding this feature. Will add  ts in new code.

Reply:
I agree adding it in the whole file will result in a large diff :slightly_smiling_face: One of the benefits behind having a TS file and a `@ts-nocheck` flag instead of just a JS file is that we can add types gradually, to some parts of the file, resulting in less work in the future and better comprehension for the PR reviewers :slightly_smiling_face: For example, I try to add type annotations for the code that I modify in TS files (even if they don't have types themselves yet) just to have that much work already done. It also helps me write code at the same time :smile: 

Anyway, it's minor, we can let it be for now. I'll resolve the comment